// office-ui-fabric-react/lib-amd/components/pickers/BasePicker.styles.js
define(["require", "exports", "../../Styling"], function (require, exports, Styling_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var GlobalClassNames = {
        root: 'ms-BasePicker',
        text: 'ms-BasePicker-text',
        itemsWrapper: 'ms-BasePicker-itemsWrapper',
        input: 'ms-BasePicker-input'
    };
    function getStyles(props) {
        var className = props.className, theme = props.theme, isFocused = props.isFocused, inputClassName = props.inputClassName;
        if (!theme) {
            throw new Error('theme is undefined or null in base BasePicker getStyles function.');
        }
        var semanticColors = theme.semanticColors;
        var inputBorder = semanticColors.inputBorder, inputBorderHovered = semanticColors.inputBorderHovered, inputFocusBorderAlt = semanticColors.inputFocusBorderAlt;
        var classNames = Styling_1.getGlobalClassNames(GlobalClassNames, theme);
        return {
            root: [classNames.root, className],
            text: [
                classNames.text,
                {
                    display: 'flex',
                    flexWrap: 'wrap',
                    alignItems: 'center',
                    boxSizing: 'border-box',
                    minWidth: 180,
                    minHeight: 30,
                    border: "1px solid " + inputBorder
                },
                !isFocused && {
                    selectors: {
                        ':hover': {
                            borderColor: inputBorderHovered
                        }
                    }
                },
                isFocused && {
                    borderColor: inputFocusBorderAlt
                }
            ],
            itemsWrapper: [
                classNames.itemsWrapper,
                {
                    display: 'flex',
                    flexWrap: 'wrap'
                }
            ],
            input: [
                classNames.input,
                {
                    height: 34,
                    border: 'none',
                    flexGrow: 1,
                    outline: 'none',
                    padding: '0 6px 0',
                    alignSelf: 'flex-end'
                },
                inputClassName
            ],
            screenReaderText: Styling_1.hiddenContentStyle
        };
    }
    exports.getStyles = getStyles;
});
